app/src/main/java/com/door43/translationstudio/newui/newtranslation/NewTargetTranslationActivity.java

```diff
@@ -345,4 +444,54 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
             }
         }
     }
+
+    @Override
+    public void onFinished(ManagedTask task) {
+        taskWatcher.stop();
+        if(task instanceof MergeTargetTranslationTask) {
+            MergeTargetTranslationTask mergeTask = (MergeTargetTranslationTask) task;
+            MergeTargetTranslationTask.Status status = mergeTask.getStatus();
+
+            int results = RESULT_ERROR;
+
+            if(MergeTargetTranslationTask.Status.MERGE_CONFLICTS == status) {
+                results = RESULT_MERGE_CONFLICT;
+            } else if(MergeTargetTranslationTask.Status.SUCCESS == status) {
+                results = RESULT_OK;
+            }
+
+            Intent data = new Intent();
+            data.putExtra(EXTRA_TARGET_TRANSLATION_ID, mergeTask.getDestinationTranslation().getId());
+            setResult(results, data);
+            finish();
+        }
+    }
+
+
+    /**
+     * for keeping track if dialog is being shown for orientation changes
+     */
+    public enum eDialogShown {
```

Comment:
@neutrinog - It's not about strings here, but converting back and forth to integers in order to save state.  Is there an easier way?  Java doesn't allow casting between enums and integers.